JamesRexMiller4/refactortractor
# Refactor classes

# What's this PR do?
-Will rewrite the User and parent Repository Class(formerly UserRepository). 
-Remove the methods from the child classes (Hydration, Sleep, Activity)
-Adds passing tests for all classes
-Amends the scripts file to call methods using the new namings
-Delete Activity class 

### This PR will NOT BREAK functionality, but will RESTORE IT.

# Where should the reviewer start?

Reviewers should start with the User and Repository Class files. The initializing order will be UserRepo(Repository) and then the User, and then the other children (Hydration, Sleep, Activity)

Once comfortable with the structure of the User and the Repository class, I would then traverse the child classes, taking note of what methods are deleted, what will be inherited, and what methods remain because they are unique to that class.

Reviewers should then hop over to the test files and look at how the tests are written to make sure no false positives are occurring.

Finally, double check the scripts.js file to make sure that all the jQuery DOM manipulations are calling the new methods correctly, and the right data is being added to the right place. 

# How should this be manually tested?

~At this time with the testing not fully rewritten and all wires not properly connected, I would not spend too much time trying to test these files.~
~Test file for Repository has been added with most recent commit~
Update:
Tests have been written for all methods of the object classes, you can explore the test file and run npm test in your terminal to test and confirm. At this time the site is still not functional, there are some issues with the fetch requests that are causing error messages to be thrown in the console.

@RayRedGoose @Garrett-Iannuzzi @brittanystoroz @khalidwilliams 

Will resolve #40 #1 #12 #13
